esp-hal/MIGRATING-0.21.md

```diff
@@ -23,13 +23,34 @@ For example:
  }
 ```
 
+## Removed `async`-specific constructors
```

Comment:
What's the rationale behind this? Every other driver uses `new` & `new_async`? If we're going to make a change like this we need to be consistent.

Reply:
I want to start working on https://github.com/esp-rs/esp-hal/issues/2321 and plan to touch up every other driver eventually

Reply:
Let's draft this until we flesh out exactly what we want to do in #2321

Reply:
Undid those bits and I'll send in a follow-up later